_docs/tasks/security/authn-policy.md

```diff
@@ -0,0 +1,300 @@
+
+```bash
+kubectl create ns foo
+kubectl apply -f <(istioctl kube-inject -f samples/httpbin/httpbin.yaml) -n foo
+kubectl apply -f <(istioctl kube-inject -f samples/sleep/sleep.yaml) -n foo
+
+kubectl create ns bar
+kubectl apply -f <(istioctl kube-inject -f samples/httpbin/httpbin.yaml) -n bar
+kubectl apply -f <(istioctl kube-inject -f samples/sleep/sleep.yaml) -n bar
+
+kubectl create ns legacy
+kubectl apply -f samples/sleep/sleep.yaml -n legacy
+
+```
+
+* Verifying setup by sending an http request (using curl command) from any sleep pod (among those in namespace `foo`, `bar` or `legacy`) to either `httpbin.foo` or `httpbin.bar`. All requests should success with http code 200.
+
+For example, here is a command to check `sleep.bar` to `httpbin.foo` reachability:
+
+```bash
+kubectl exec $(kubectl get pod -l app=sleep -n bar -o jsonpath={.items..metadata.name}) -c sleep -n bar -- curl http://httpbin.foo:8000/ip -s -o /dev/null -w "%{http_code}\n"
+```
+```
+200
+```
+
+Conveniently, this one-liner command iterates through all combinations:
+
+```bash
+for from in "foo" "bar" "legacy"; do for to in "foo" "bar"; do kubectl exec $(kubectl get pod -l app=sleep -n ${from} -o jsonpath={.items..metadata.name}) -c sleep -n ${from} -- curl http://httpbin.${to}:8000/ip -s -o /dev/null -w "sleep.${from} to httpbin.${to}: %{http_code}\n"; done; done
+```
+
+```
+sleep.foo to httpbin.foo: 200
+sleep.foo to httpbin.bar: 200
+sleep.bar to httpbin.foo: 200
+sleep.bar to httpbin.bar: 200
+sleep.legacy to httpbin.foo: 200
+sleep.legacy to httpbin.bar: 200
+```
+
+
+* Also verify that there are no authencation policy in the system
+
+```bash
+kubectl get policies.authentication.istio.io -n foo
+
+kubectl get policies.authentication.istio.io -n bar
+```
+
+```
+No resources found.
+```
+
+## Enable mTLS for all services in namespace `foo`
+
+Run this command to set namespace-level policy for namespace `foo`.
+
+```bash
+cat <<EOF | istioctl create -f -
+apiVersion: "authentication.istio.io/v1alpha1"
+kind: "Policy"
+metadata:
+  name: "enable-mtls"
+  namespace: "foo"
+spec:
+  peers:
+  - mtls:
+EOF  
+```
+
+And verify policy is added:
+```bash
+kubectl get policies.authentication.istio.io -n foo
+```
+
+```
+NAME          AGE
+enable-mtls   1m
+```
+
+Run the same testing command above. We should see request from `sleep.legacy` to `httpbin.foo` start to fail, as the result of enabling mTLS for `httpbin.foo` but `sleep.legacy` doesn't have sidecar to support it. On the other hand, for clients with sidecar (`sleep.foo` and `sleep.bar`), Istio automatically configures them to using mTLS where talking `http.foo`, so they continue to work. Also, requests to `httpbin.bar` are not affected as policy is effective on `foo` namespace only.
```

Comment:
as policy is -> as the policy is

Reply:
talking `http.foo` -> talking to `http.foo`

Reply:
Done